projects/observability/src/shared/services/global-csv-download-table/global-csv-download-table.service.ts

```diff
@@ -0,0 +1,107 @@
+import { Injectable } from '@angular/core';
+import { Dictionary } from '@hypertrace/common';
+import {
+  CsvDownloadFileConfig,
+  FileDownloadService,
+  TableColumnConfig,
+  TableDataResponse,
+  TableDataSource,
+  TableRow,
+  TableSortDirection
+} from '@hypertrace/components';
+import { isNil } from 'lodash-es';
+import { Observable, throwError } from 'rxjs';
+import { map, switchMap } from 'rxjs/operators';
+
+@Injectable({ providedIn: 'root' })
+export class GlobalCsvDownloadTableService {
```

Comment:
`Table` in the name is not required. We could use this to download data from non-table consumers as well